app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayImageActivity.kt

```diff
@@ -1,28 +1,67 @@
 package com.github.HumanLearning2021.HumanLearningApp.view
 
+import android.content.Intent
 import android.os.Bundle
 import android.view.View
 import android.widget.TextView
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
 import com.github.HumanLearning2021.HumanLearningApp.R
 import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseManagement
+import kotlinx.coroutines.launch
 
 class DisplayImageActivity : AppCompatActivity() {
+
+    private lateinit var picture: CategorizedPicture
+    private lateinit var datasetId: String
+    private val staticDBManagement = DummyDatabaseManagement.staticDummyDatabaseManagement
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_display_image)
 
-        val image: CategorizedPicture =
-            intent.getParcelableExtra("display_image_image")!!
-
-        findViewById<TextView>(R.id.display_image_viewCategory).text = image.category.name
-        image.displayOn(this, findViewById(R.id.display_image_viewImage))
+        picture =
+            intent.getParcelableExtra("single_picture")!!
+        datasetId = intent.getStringExtra("dataset_id")!!
+        findViewById<TextView>(R.id.display_image_viewCategory).text = picture.category.name
+        picture.displayOn(this, findViewById(R.id.display_image_viewImage))
     }
 
     fun removePicture(view: View) {
-        //TODO: Implement : remove picture from the list and send intent to the previous activity with the updated list
+        var noMorePicturesInThisCategory = false
+        lifecycleScope.launch {
+            staticDBManagement.removePicture(picture)
+            if (staticDBManagement.getAllPictures(picture.category).isEmpty()) {
+                noMorePicturesInThisCategory = true
+            }
+        }
         Toast.makeText(this, getText(R.string.picturehasbeenremoved), Toast.LENGTH_SHORT)
             .show()
+        if (noMorePicturesInThisCategory) {
+            launchDisplayDatasetActivity()
+        } else {
+            launchDisplayImageSetActivity()
+        }
```

Comment:
Careful! This is a race condition!

Reply:
hm... You're totally right. Do you have any idea to prevent this race condition by keeping the same idea?